test/intest1.F90

```diff
@@ -0,0 +1,100 @@
+
+  print *, 'Testing reading IN_1, CRBMG with OPENBF IO = SEC3'
+
+  ! Open the test file.
+  call cobfl(filnam, filost)
+
+  ! Specify format of section 1 date/time when reading.
+  call datelen(10)
+
+  ! Specify the use of section 3 decoding.
+  open (unit = 11, file = '/dev/null')
+  call openbf(11, 'SEC3', 11)
+
+  ! Specify location of master BUFR tables.
+  call mtinfo('../tables', 90, 91)
+
+  ! Read a BUFR message from the test file into a memory array.
+  call crbmg(bfmg, mxbf, nbyt, ierr)
+  if (ierr .ne. 0) stop 2
+
+  ! Read and check some values from Section 1.
+  if ((iupbs01(ibfmg, 'MTYP') .ne. 2) .or. &
+       (iupbs01(ibfmg, 'MTV') .ne. 14) .or. (iupbs01(ibfmg, 'LENM') .ne. 4169)) stop 3
+
+  ! Read and check some values from Section 3. 
+  if ((iupbs3(ibfmg, 'NSUB') .ne. 1) .or. &
+       (iupbs3(ibfmg, 'ICMP') .ne. 0)) stop 4
+
+  ! Read and check some data descriptors from Section 3.
+  call upds3(ibfmg, mxds3, cds3, nds3)
+  IF (nds3 .ne. 8 .or. cds3(1) .ne. '309052' .or. cds3(5) .ne. '002095') stop 5
+
+  ! Pass the BUFR message from the memory array into the library.
+  call readerme(ibfmg, 11, cmgtag, imgdt, ierme)
+  if (ierme .ne. 0 .or. cmgtag .ne. 'MSTTB001') stop 6
+
+  ! Get and check the element names and units associated with some
+  ! Table B mnemonics.
+  call nemdefs(11, 'VSIGX', celem(1), cunit(1), ierndv)
+  call nemdefs(11, 'SMID', celem(2), cunit(2), iernds)
+  if (ierndv .ne. 0 .or. iernds .ne. 0 .or. celem(1)(1:40) .ne. &
+       'Extended vertical sounding significance ' .or. celem(2)(1:39) .ne. &
+       'Ship or mobile land station identifier ' .or. cunit(1)(1:12) .ne. &
+       'FLAG TABLE  ' .or. cunit(2)(1:10) .ne. 'CCITT IA5 ') stop 7
+
+  ! Read and check the Section 1 date-time.
+  if (imgdt .ne. 2012093012) stop 8
+
+  ! Read a data subset from the BUFR message.
+  if (ireadsb(11) .ne. 0 ) stop 9
+
+  ! Get and check the parent of a Table B mnemonic.
+  call gettagpr(11, 'PRLC', 192, tagpr, iertgp)
+  if (iertgp .ne. 0 .or. tagpr .ne. 'WSPLRAOB') stop 10
+
+  ! Read and check some data values.
+  call ufbint(11, r8arr, MXR8PM, MXR8LV, nr8lv, 'CLONH A4ME HSMSL QCEVR')
+  IF (nr8lv .ne. 1 .or. NINT(r8arr(1,1)*100000) .ne. 10388797 .or. &
+       NINT(r8arr(2,1)) .ne. 7 .or. NINT(r8arr(3,1)) .ne. 14 .or. &
+       ibfms(r8arr(4,1)) .ne. 1) stop 11
+
+  ! Read and check a sequence of data values.
+  call ufbseq(11, r8arr, MXR8PM, MXR8LV, nr8lv, 'TDWPRAOB')
+  IF (nr8lv .ne. 191 .or. NINT(r8arr(8,3)*100) .ne. 29416 .or. &
+       NINT(r8arr(10,11)*10) .ne. 55 .or. NINT(r8arr(2,12)) .ne. 2048 .or. &
+       NINT(r8arr(5,67)*100000) .ne. -1167 .or. NINT(r8arr(1,186)) .ne. 2523) stop 12
+
+  ! Read and check some long character strings.
+  call readlc(11, smidstg, 'SMID')
+  call readlc(11, softvstg, 'SOFTV')
+  IF (smidstg(7:9) .ne. 'UAO' .or. softvstg(5:12) .ne. '5.8.5.10') stop 13
+
+  ! Close the test file.
+  call ccbfl()
+
+  print *, 'SUCCESS!'
+end program intest1
```

Comment:
Just so I understand, and since this is no longer being run by the test_wrapper_IN.sh script, the idea is that if it makes it all the way through to the end, then it will end up returning with a zero return code by default, but if it hits any of the prior `stop` statements then it will end up with the associated positive number as a return code, and in turn that will definitely get caught by the 
CI runner and the test will fail, correct?

Reply:
Also, I noticed you started with `stop 2` as your first error exit rather than `stop 1`.  Any particular reason for that?  Not a big deal (assuming I'm correctly understanding that anything other than zero automatically gets trapped as a failure?), but more just curious than anything?

Reply:
Correct. If the program reaches "end program" then it will return a 0.

If any stop is called, it will return non-zero. From the number of the stop, you can tell exactly where the failure is, if you use a different number for each stop command.

Why not start with 1? I don't know. Good question. I just started with 2 for some reason.